CMSgov/eAPD
# bbrooks/3168-Date_Not_Updating

resolves # 3168

**Description-**
Added strict validation to stateDateToDisplay to resolve the dates not updating properly when updated to an incorrect value and sometimes not displaying the "Invalid Date" message.

**This pull request changes...**

- The date field will update, even if provided an invalid date
- The summary, table, or export view, etc. show show invalid date when given an invalid date

**This pull request also touches…**

- Date related stuff

**This pull request was tested in the follow ways…**

- list of frontend cases
- list of backend cases

**Steps to manually verify this change...**

1. Log into test instance
2. Navigate to a place where a date is set (Activities > Activities 1: Program Administration > Activity overview > Activity Schedule > Start date for example)
3. Set the date to an invalid date.
4. Click continue at the bottom of the page
5. Navigate to a place where the date is displayed (Executive Summary > Executive Summary > Activity 1: Program Administration > Start date - End date for example)
6. View the beautiful "Invalid date" entries for the date

### This pull request is ready to review when...


### This pull request can be merged when…
